crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -2740,6 +2750,35 @@ impl ClosureReplacer<'_> {
 }
 
 impl VisitMut for ClosureReplacer<'_> {
+    fn visit_mut_stmt(&mut self, stmt: &mut Stmt) {
+        stmt.visit_mut_children_with(self);
+
+        // Replace super calls in hoisted static class methods with empty
+        // statements so that we can remove them later to ensure that the
+        // resulting JS is still valid. Before this, we have already emitted an
+        // error for the super call.
+        if let Stmt::Expr(ExprStmt {
+            expr:
+                box Expr::Call(CallExpr {
+                    callee: Callee::Expr(box Expr::SuperProp(SuperPropExpr { .. })),
+                    ..
+                }),
+            ..
+        }) = stmt
+        {
+            stmt.take();
+        }
+    }
```

Comment:
i'm confused, why do we need this if we've already emitted an error? is it to prevent getting a secondary "`super` outside of classes is invalid" syntax error because we hoisted the code out `super`? if so, can't we just avoid hoisting the method at all?

Reply:
> if so, can't we just avoid hoisting the method at all?

I like the idea; addressed in #73189.